R/pairwise-comparisons.R

```diff
@@ -1,24 +1,53 @@
-#' @title Do Pairwise Comparisons of Scores
+#' @title Obtain Pairwise Comparisons Between Models
 #'
 #' @description
 #'
-#' Compute relative scores between different models making pairwise
-#' comparisons. Pairwise comparisons are a sort of pairwise tournament where all
+#' Compare scores obtained by different models in a pairwise tournament. All
 #' combinations of two models are compared against each other based on the
 #' overlapping set of available forecasts common to both models.
-#' Internally, a ratio of the mean scores of both models is computed.
-#' The relative score of a model is then the geometric mean of all mean score
-#' ratios which involve that model. When a baseline is provided, then that
-#' baseline is excluded from the relative scores for individual models
-#' (which therefore differ slightly from relative scores without a baseline)
-#' and all relative scores are scaled by (i.e. divided by) the relative score of
-#' the baseline model.
-#' Usually, the function input should be unsummarised scores as
-#' produced by [score()].
-#' Note that the function internally infers the *unit of a single forecast* by
-#' determining all columns in the input that do not correspond to metrics
-#' computed by [score()]. Adding unrelated columns will change results in an
-#' unpredictable way.
+#'
+#' The input should be a `scores` object as produced by [score()]. Note that
+#' adding additional unrelated columns can unpredictably change results.
+#' The output of the pairwise comparisons is a set of mean score ratios,
+#' relative skill scores and p-values.
```

Comment:
maybe also add a description to @return? Something like
```r
#' @return A data.table with the results of pairwise comparisons containing the
#' mean score ratios (`mean_scores_ratio`), unadjusted (`pval`) and adjusted
#' (`adj_pval`) p-values, and relative skill values of each model
#' (`..._relative_skill`). If a baseline model is given then the scaled
#' relative skill is also reported as (`..._scaled_relative_skill`).
```

Reply:
Done